drivers/hmis/app/graphql/types/hmis_schema/submit_form_result.rb

```diff
@@ -23,9 +23,9 @@ class HmisSchema::SubmitFormResult < Types::BaseUnion
 
     def self.resolve_type(object, _context)
       config = Hmis::Form::Definition::FORM_ROLE_CONFIG.find do |_, value|
-        value[:class_name] == object.class.name
+        value[:class_name] == object.class.sti_name
       end&.last
-      raise "#{object.class.name} is not a valid response type" unless config.present?
+      raise "#{object.class.sti_name} is not a valid response type" unless config.present?
```

Comment:
Might make more sense to leave this as name, since it's displayed to the developer, not using it to match something in the db.